docs/src/rules/no-restricted-imports.md

```diff
@@ -118,6 +118,19 @@ Pattern matches can restrict specific import names only, similar to the `paths`
 }]
 ```
 
+Regex patterns can also be used to restrict specific import Name:
+
+```json
+"no-restricted-imports": ["error", {
+    "patterns": [{
+      "group": ["import-foo/*"],
+      "importNamePatterns": "^foo",
+    }]
+}]
+```
+
+**Note:** In patterns array `importNames` and `importNamePatterns` can not be used together. If they are used together then the `importNames` will be considered.
```

Comment:
We didn't discuss this on the issue, but I don't think there's a reason not to allow both to be used at the same time.

Reply:
Agreed, 
Then can i get an overview about how they should work together and what if they are referring the same import name.

Reply:
@mdjermanovic is away for the rest of the week, but here's my thinking: you should first check the name against all `importNames` and if there aren't any matches, then check against `importNamePattern`.

Reply:
Sounds good to me, working on this approach.